data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/tekuv1/beacon/GetAllBlocksAtSlot.java

```diff
@@ -77,27 +103,48 @@ public GetAllBlocksAtSlot(
       })
   @Override
   public void handle(@NotNull final Context ctx) throws Exception {
-    final Map<String, String> pathParamMap = ctx.pathParamMap();
-    ctx.header(Header.CACHE_CONTROL, CACHE_NONE);
+    adapt(ctx);
+  }
+
+  @Override
+  public void handleRequest(RestApiRequest request) throws JsonProcessingException {
+    request.header(Header.CACHE_CONTROL, CACHE_NONE);
+
+    final UInt64 slot = request.getPathParameter(SLOT_PARAMETER);
+    final SafeFuture<AllBlocksAtSlotData> future = chainDataProvider.getAllBlocksAtSlot(slot);
+
+    request.respondAsync(
+        future.thenApply(
+            result -> {
+              if (result.getBlocks().isEmpty()) {
+                return AsyncApiResponse.respondWithError(SC_NOT_FOUND, "Blocks not found: " + slot);
+              }
+
+              return AsyncApiResponse.respondOk(result);
+            }));
+  }
+
+  public static SerializableTypeDefinition<AllBlocksAtSlotData> getResponseType(
+      final SchemaDefinitionCache schemaDefinitionCache) {
+    final SerializableOneOfTypeDefinition<BeaconBlock> messageType =
+        getSchemaDefinitionForAllMilestones(
+            schemaDefinitionCache,
+            "BeaconBlock",
+            SchemaDefinitions::getBeaconBlockSchema,
+            (beaconBlock, milestone) ->
+                schemaDefinitionCache.milestoneAtSlot(beaconBlock.getSlot()).equals(milestone));
+
+    final SerializableTypeDefinition<SignedBeaconBlock> dataType =
+        SerializableTypeDefinition.object(SignedBeaconBlock.class)
+            .withField("message", messageType, SignedBeaconBlock::getMessage)
+            .withField("signature", SIGNATURE_TYPE, SignedBeaconBlock::getSignature)
+            .withField("root", BYTES32_TYPE, SignedBeaconBlock::getRoot)
+            .build();
```

Comment:
this is going to be more like `v2.beacon.GetBlock.GetResponseType`

Reply:
Not quite because this one needs a `"root"` field, while those ones were just the same as the `SignedBeaconBlock` schema

Reply:
oh :/ how annoying